src/indexer/segment_writer.rs

```diff
@@ -248,7 +248,7 @@ impl SegmentWriter {
                 FieldType::Date(_) => {
                     for value in values {
                         let date_val = value.as_date().ok_or_else(make_schema_error)?;
-                        term_buffer.set_u64(date_val.to_u64());
+                        term_buffer.set_u64(date_val.to_u64(None));
```

Comment:
See discussion in json_term_writer.
We want some truncation here.

I suggest we truncate to the second all of the time.